include/highfive/bits/H5Slice_traits.hpp

```diff
@@ -49,6 +49,122 @@ class ElementSet {
     friend class SliceTraits;
 };
 
+class HyperSlab {
+  public:
+    struct Select {
+        Select() = default;
+
+        Select(std::vector<size_t> offset_, std::vector<size_t> count_, std::vector<size_t> stride_)
+            : offset(offset_), count(count_), stride(stride_)
+        {}
+
+        std::vector<size_t> offset;
+        std::vector<size_t> count;
+        std::vector<size_t> stride;
+    };
+
+    HyperSlab() {
+        selects.emplace_back(Select{}, Op::None);
+    };
+
+    HyperSlab(Select sel) {
+        selects.emplace_back(sel, Op::Set);
+    }
+
+    HyperSlab operator|(const Select& sel) const {
+        auto ret = *this;
+        ret.selects.emplace_back(sel, Op::Or);
+        return ret;
+    }
+
+    HyperSlab& operator|=(const Select& sel) {
+        selects.emplace_back(sel, Op::Or);
+        return *this;
+    }
+
+    HyperSlab operator&(const Select& sel) const {
+        auto ret = *this;
+        ret.selects.emplace_back(sel, Op::And);
+        return ret;
+    }
+
+    HyperSlab& operator&=(const Select& sel) {
+        selects.emplace_back(sel, Op::And);
+        return *this;
+    }
+
+    HyperSlab operator^(const Select& sel) const {
+        auto ret = *this;
+        ret.selects.emplace_back(sel, Op::Xor);
+        return ret;
+    }
+
+    HyperSlab& operator^=(const Select& sel) {
+        selects.emplace_back(sel, Op::Xor);
+        return *this;
+    }
+
+    void apply(const DataSpace& space) {
```

Comment:
I should make a copy of this DataSpace and return it, right?
From this new DataSpace we can retrieve new dims for the `Selection` class.